Primitive object creation with Is Sentence and attributes causes exception

## Example

```markdown
# My First Scenario

Foo is an Object with name bar.
```

## Output

```
// failed to execute phase 'NameResolver' due to exception:
// java.lang.NullPointerException
// 	at org.fulib.scenarios.visitor.resolve.DeclResolver.resolveAttribute(DeclResolver.java:216)
// 	at org.fulib.scenarios.visitor.resolve.DeclResolver.resolveAttributeOrAssociation(DeclResolver.java:208)
// 	at org.fulib.scenarios.visitor.resolve.SentenceResolver.resolveSimpleHasNamedExpr(SentenceResolver.java:431)
// 	at org.fulib.scenarios.visitor.resolve.SentenceResolver.resolveHasNamedExpr(SentenceResolver.java:384)
// 	at org.fulib.scenarios.visitor.resolve.ExprResolver.visit(ExprResolver.java:129)
// 	at org.fulib.scenarios.visitor.resolve.ExprResolver.visit(ExprResolver.java:43)
// 	at org.fulib.scenarios.ast.expr.call.CreationExpr.accept(CreationExpr.java:25)
// 	at org.fulib.scenarios.visitor.resolve.SentenceResolver.visit(SentenceResolver.java:452)
// 	at org.fulib.scenarios.visitor.resolve.SentenceResolver.visit(SentenceResolver.java:38)
// 	at org.fulib.scenarios.ast.sentence.IsSentence.accept(IsSentence.java:20)
// 	at org.fulib.scenarios.visitor.resolve.SentenceResolver.visit(SentenceResolver.java:54)
// 	at org.fulib.scenarios.visitor.resolve.SentenceResolver.visit(SentenceResolver.java:38)
// 	at org.fulib.scenarios.ast.sentence.SentenceList.accept(SentenceList.java:20)
// 	at org.fulib.scenarios.visitor.resolve.NameResolver.visit(NameResolver.java:131)
// 	at org.fulib.scenarios.visitor.resolve.NameResolver.visit(NameResolver.java:19)
// 	at org.fulib.scenarios.ast.Scenario.accept(Scenario.java:24)
// 	at org.fulib.scenarios.visitor.resolve.NameResolver.visit(NameResolver.java:105)
// 	at org.fulib.scenarios.visitor.resolve.NameResolver.visit(NameResolver.java:19)
// 	at org.fulib.scenarios.ast.ScenarioFile.accept(ScenarioFile.java:30)
// 	at org.fulib.scenarios.visitor.resolve.NameResolver.visit(NameResolver.java:58)
// 	at org.fulib.scenarios.visitor.resolve.NameResolver.visit(NameResolver.java:19)
// 	at org.fulib.scenarios.ast.ScenarioGroup.accept(ScenarioGroup.java:26)
// 	at org.fulib.scenarios.visitor.resolve.NameResolver.lambda$visit$0(NameResolver.java:43)
// 	(13 internal calls)
// 	at org.fulib.scenarios.visitor.resolve.NameResolver.visit(NameResolver.java:43)
// 	at org.fulib.scenarios.ast.CompilationContext.accept(CompilationContext.java:24)
// 	(3 internal calls)
// 	at org.fulib.webapp.tool.Tools.scenarioc(Tools.java:83)
// 	at org.fulib.webapp.tool.Tools.genCompileRun(Tools.java:161)
// 	at org.fulib.webapp.tool.RunCodeGen.run(RunCodeGen.java:148)
// 	at org.fulib.webapp.tool.RunCodeGen.handle(RunCodeGen.java:50)
// 	(21 internal calls)
// failed to execute phase 'CodeGenerator' due to exception:
// java.lang.NullPointerException
// 	(12 internal calls)
// 	at org.fulib.scenarios.visitor.codegen.CodeGenerator.visit(CodeGenerator.java:51)
// 	at org.fulib.scenarios.ast.CompilationContext.accept(CompilationContext.java:24)
// 	(3 internal calls)
// 	at org.fulib.webapp.tool.Tools.scenarioc(Tools.java:83)
// 	at org.fulib.webapp.tool.Tools.genCompileRun(Tools.java:161)
// 	at org.fulib.webapp.tool.RunCodeGen.run(RunCodeGen.java:148)
// 	at org.fulib.webapp.tool.RunCodeGen.handle(RunCodeGen.java:50)
// 	(21 internal calls)
// Caused by: java.lang.NullPointerException
// 	at org.fulib.scenarios.visitor.resolve.SymbolCollector.getRoots(SymbolCollector.java:80)
// 	at org.fulib.scenarios.visitor.resolve.SymbolCollector.getRoots(SymbolCollector.java:49)
// 	at org.fulib.scenarios.visitor.resolve.SymbolCollector.getRoots(SymbolCollector.java:27)
// 	at org.fulib.scenarios.visitor.codegen.CodeGenerator.addDiagramSentences(CodeGenerator.java:338)
// 	at org.fulib.scenarios.visitor.codegen.CodeGenerator.visit(CodeGenerator.java:313)
// 	at org.fulib.scenarios.visitor.codegen.CodeGenerator.visit(CodeGenerator.java:31)
// 	at org.fulib.scenarios.ast.ScenarioFile.accept(ScenarioFile.java:30)
// 	at org.fulib.scenarios.visitor.codegen.CodeGenerator.populateTests(CodeGenerator.java:163)
// 	at org.fulib.scenarios.visitor.codegen.CodeGenerator.visit(CodeGenerator.java:148)
// 	at org.fulib.scenarios.visitor.codegen.CodeGenerator.visit(CodeGenerator.java:31)
// 	at org.fulib.scenarios.ast.ScenarioGroup.accept(ScenarioGroup.java:26)
// 	at org.fulib.scenarios.visitor.codegen.CodeGenerator.lambda$visit$0(CodeGenerator.java:55)
// 	(9 internal calls)
```

--- a/src/main/java/org/fulib/scenarios/visitor/resolve/ExprResolver.java
+++ b/src/main/java/org/fulib/scenarios/visitor/resolve/ExprResolver.java
@@ -121,12 +121,30 @@ public enum ExprResolver implements Expr.Visitor<Scope, Expr>
    @Override
    public Expr visit(CreationExpr creationExpr, Scope par)
    {
-      creationExpr.setType(creationExpr.getType().accept(TypeResolver.INSTANCE, par));
-      final ClassDecl classDecl = creationExpr.getType().accept(ExtractClassDecl.INSTANCE, null);
+      final Type receiverType = creationExpr.getType().accept(TypeResolver.INSTANCE, par);
+      creationExpr.setType(receiverType);
+
+      if (receiverType == PrimitiveType.ERROR)
+      {
+         // recover from previous error
+         return creationExpr;
+      }
+
+      final ClassDecl receiverClass = receiverType.accept(ExtractClassDecl.INSTANCE, null);
+      if (receiverClass == null)
+      {
+         if (!creationExpr.getAttributes().isEmpty())
+         {
+            par.report(
+               error(creationExpr.getPosition(), "create.subject.primitive.attributes", receiverType.getDescription()));
+         }
+
+         return creationExpr;
+      }
 
       for (final NamedExpr namedExpr : creationExpr.getAttributes())
       {
-         SentenceResolver.resolveHasNamedExpr(namedExpr, classDecl, par);
+         SentenceResolver.resolveHasNamedExpr(namedExpr, receiverClass, par);
       }
       return creationExpr;
    }

--- a/src/main/resources/org/fulib/scenarios/diagnostic/messages.properties
+++ b/src/main/resources/org/fulib/scenarios/diagnostic/messages.properties
@@ -63,6 +63,7 @@ descriptor.multi.indefinite.deprecated=the '<type>s <names>' syntax is deprecate
   write 'the %s %s' instead
 
 has.subject.primitive=invalid has sentence - subject has primitive type '%s'
+create.subject.primitive.attributes=primitive type '%s' cannot be instantiated with attributes
 write.target.invalid=invalid write target - cannot write into %s
 write.target.list=cannot write into %s - only single-valued attributes and associations are allowed
 assign.type=cannot assign expression of type '%s' to variable '%s' of type '%s'

--- a/src/test/invalid_scenarios/lang/There.md
+++ b/src/test/invalid_scenarios/lang/There.md
@@ -61,3 +61,8 @@ perhaps this name was inferred from the first attribute and you need to give thi
 error: invalid redeclaration of 'invalidRedeclarations' [variable.redeclaration]
 perhaps this name was inferred from the first attribute and you need to give this object an explicit name?
 -->
+
+(#190) O3 is an Object with name o3.
+<!--            ^
+error: primitive type 'Object' cannot be instantiated with attributes [create.subject.primitive.attributes]
+-->

